src/main/java/io/sdkman/broker/download/CandidateDownloadHandler.java

```diff
@@ -47,12 +58,44 @@ public void handle(Context ctx) {
                                             downloadResolver.resolve(downloads, p.name())
                                                     .ifPresentOrElse(v -> {
                                                         audit(details, p.id(), v.platform());
+                                                        writeChecksumHeaders(ctx, v.checksums());
                                                         ctx.redirect(302, v.url());
                                                     }, clientError(ctx, 404))),
                             clientError(ctx, 400));
         }, clientError(ctx, 404));
     }
 
+    private void writeChecksumHeaders(Context ctx,
+        Option<scala.collection.immutable.Map<String, String>> checksums) {
+        if(checksums == null || checksums.isEmpty()){
+            return;
+        }
+
+       Comparator<String> algoComparator =  Comparator.comparing((Function<String, Integer>) algorithm -> {
+            if (MD5.id().equals(algorithm)) {
```

Comment:
Java interop with scala's traits/case objects isn't great, hence this ugly `if` block (wouldn't be nice to write this as a simple switch with pattern matching?)

Reply:
Totally agree. I still have it on my long todo list to rewrite this service in Scala/akka-http or Kotlin/http4k at some point. Some many things to do, so little time :smile: